src/pymor/vectorarrays/block.py

```diff
@@ -24,6 +24,7 @@ class BlockVectorArray(VectorArrayInterface):
     def __init__(self, blocks, space):
         self._blocks = tuple(blocks)
         self.space = space
+        self.dtype = reduce(np.promote_types, (v.dtype for v in self._blocks))
```

Comment:
Is it ok to use something like `self.dtype = tuple(v.dtype for v in self._blocks)`? This would better represent the data type of the vectors, but I also can't think of a use case where a block vector would have vectors of different dtypes...

Reply:
One could not make a usable `ones` with that tuple dtype.

Reply:
Why not? Every vector could have its own type of 1, right?

Reply:
Already has, yes. "Usable" is not the right word I guess. But the tuple dtype would lead to the vector data being block-interleaved in memory instead of block-consecutive. You'd have to swap all the algebra ops and probably end up with worse performance.

Reply:
I'm confused. Do vectorarrays even need to be consecutive in memory when inside a `BlockVectorArray`?

Reply:
No. I'm confused. I was in an entirely different software in my head. Forgot everything I answered, sorry.

Using the tuple dtype is technically possible.